website/docs/src/markdown-conversion/unified-plugins/rehype-label.js

```diff
@@ -0,0 +1,34 @@
+import { visit } from 'unist-util-visit';
+
+export default function addLabelTag() {
+  return function transformer(tree) {
+    visit(tree, ['comment'], (node, index, parent) => {
+      const trimmedValue = node.value.trim();
+      const match = trimmedValue.match(/^label-(\w+):\s*(.*)$/);
+
+      if (match) {
+        const className = match[1];
+        const labelText = match[2];
+        transformToLabel(node, index, parent, className, labelText);
+      }
+    });
+  };
+
+  function transformToLabel(node, index, parent, className, labelText) {
```

Comment:
jsdoc here as well. 

would be helpful to also visualise input/outputs what the HTML looks like with a very basic example too.